crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -818,6 +818,16 @@ pub fn comparator_operand_type_rules(
 
             Ok(Bool(CompTime::No(Some(op.location.span))))
         }
+        (TypeVariable(l_binding), TypeVariable(r_binding)) => {
+            if let TypeBinding::Bound(l_link) = &*l_binding.borrow() {
+                if let TypeBinding::Bound(r_link) = &*r_binding.borrow() {
+                    return comparator_operand_type_rules(l_link, r_link, op, errors);
```

Comment:
I don't see how this new case is ever triggered as a type variable on either side should always be matched on line 798

Reply:
Yes clippy is telling the same, but without this it was complaining it could not compare u8 and u8. I suppose there have been some update in master. Now I have the following error: `error: Integer cannot be used with type u8`.
The `comparator_operand_type_rules` does not support all cases. Can you fix it?

Reply:
Yes, but lets file/fix this as a separate bug if possible

Reply:
OK, I will remove this change from the PR and create an issue. The PR will need the fix else the unit test that I added is not working.

Reply:
Done, I created issue #788